File: ai-code-frontend/src/services/backend/aiModelConfigController.ts
import request from '@/utils/request';

export async function addAiModelConfig(body: API.AiModelConfig, options?: { [key: string]: any }) {
  return request.post<any, API.BaseResponseLong>('/aiModel/add', body, options);
}

export async function updateAiModelConfig(body: API.AiModelConfig, options?: { [key: string]: any }) {
  return request.post<any, API.BaseResponseBoolean>('/aiModel/update', body, options);
}

export async function updateModelStatus(body: API.AiModelConfig, options?: { [key: string]: any }) {
  return request.post<any, API.BaseResponseBoolean>('/aiModel/updateStatus', body, options);
}

export async function deleteAiModelConfig(body: API.DeleteRequest, options?: { [key: string]: any }) {
  return request.post<any, API.BaseResponseBoolean>('/aiModel/delete', body, options);
}

export async function getAiModelConfig(id: string, options?: { [key: string]: any }) {
  return request.get<any, API.BaseResponseAiModelConfig>(`/aiModel/get/${id}`, options);
}

export async function listAiModelConfig(params: {
  provider?: string;
  tier?: string;
  current?: number;
  pageSize?: number;
}, options?: { [key: string]: any }) {
  return request.get<any, API.BaseResponsePageAiModelConfig>('/aiModel/list', { params, ...options });
}

export async function getEnabledModels(options?: { [key: string]: any }) {
  return request.get<any, API.BaseResponseListAiModelConfig>('/aiModel/enabled', options);
}

File: src/main/java/com/mashang/aicode/web/controller/AiModelConfigController.java
package com.mashang.aicode.web.controller;

import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper;
import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
import com.mashang.aicode.web.annotation.AuthCheck;
import com.mashang.aicode.web.common.BaseResponse;
import com.mashang.aicode.web.common.ResultUtils;
import com.mashang.aicode.web.constant.UserConstant;
import com.mashang.aicode.web.exception.ErrorCode;
import com.mashang.aicode.web.exception.ThrowUtils;
import com.mashang.aicode.web.model.entity.AiModelConfig;
import com.mashang.aicode.web.service.AiModelConfigService;
import jakarta.annotation.Resource;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/aiModel")
public class AiModelConfigController {

    @Resource
    private AiModelConfigService aiModelConfigService;

    @PostMapping("/add")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Long> addAiModelConfig(@RequestBody AiModelConfig aiModelConfig) {
        ThrowUtils.throwIf(aiModelConfig == null, ErrorCode.PARAMS_ERROR);
        boolean result = aiModelConfigService.save(aiModelConfig);
        ThrowUtils.throwIf(!result, ErrorCode.OPERATION_ERROR);
        return ResultUtils.success(aiModelConfig.getId());
    }

    @PostMapping("/update")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Boolean> updateAiModelConfig(@RequestBody AiModelConfig aiModelConfig) {
        ThrowUtils.throwIf(aiModelConfig == null || aiModelConfig.getId() == null, ErrorCode.PARAMS_ERROR);
        boolean result = aiModelConfigService.updateById(aiModelConfig);
        ThrowUtils.throwIf(!result, ErrorCode.OPERATION_ERROR);
        return ResultUtils.success(true);
    }

    @PostMapping("/updateStatus")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Boolean> updateModelStatus(@RequestBody AiModelConfig aiModelConfig) {
        ThrowUtils.throwIf(aiModelConfig == null || aiModelConfig.getId() == null, ErrorCode.PARAMS_ERROR);
        boolean result = aiModelConfigService.updateModelStatus(aiModelConfig.getId(), aiModelConfig.getIsEnabled());
        ThrowUtils.throwIf(!result, ErrorCode.OPERATION_ERROR);
        return ResultUtils.success(true);
    }

    @PostMapping("/delete")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Boolean> deleteAiModelConfig(@RequestBody AiModelConfig aiModelConfig) {
        ThrowUtils.throwIf(aiModelConfig == null || aiModelConfig.getId() == null, ErrorCode.PARAMS_ERROR);
        boolean result = aiModelConfigService.removeById(aiModelConfig.getId());
        ThrowUtils.throwIf(!result, ErrorCode.OPERATION_ERROR);
        return ResultUtils.success(true);
    }

    @GetMapping("/get/{id}")
    public BaseResponse<AiModelConfig> getAiModelConfig(@PathVariable Long id) {
        ThrowUtils.throwIf(id == null, ErrorCode.PARAMS_ERROR);
        AiModelConfig aiModelConfig = aiModelConfigService.getById(id);
        ThrowUtils.throwIf(aiModelConfig == null, ErrorCode.NOT_FOUND_ERROR);
        return ResultUtils.success(aiModelConfig);
    }

    @GetMapping("/list")
    public BaseResponse<Page<AiModelConfig>> listAiModelConfig(
            @RequestParam(required = false) String provider,
            @RequestParam(required = false) String tier,
            @RequestParam(required = false) Integer current,
            @RequestParam(required = false) Integer pageSize) {
        QueryWrapper<AiModelConfig> queryWrapper = new QueryWrapper<>();
        if (provider != null) {
            queryWrapper.eq("provider", provider);
        }
        if (tier != null) {
            queryWrapper.eq("tier", tier);
        }
        queryWrapper.eq("is_delete", 0);
        queryWrapper.orderByAsc("sort_order");

        if (current == null || current < 1) {
            current = 1;
        }
        if (pageSize == null || pageSize < 1) {
            pageSize = 10;
        }

        Page<AiModelConfig> page = aiModelConfigService.listAiModelConfigByPage(queryWrapper, current, pageSize);
        return ResultUtils.success(page);
    }

    @GetMapping("/enabled")
    public BaseResponse<List<AiModelConfig>> getEnabledModels() {
        List<AiModelConfig> models = aiModelConfigService.getEnabledModels();
        return ResultUtils.success(models);
    }
}
